src/state/slices/opportunitiesSlice/constants.ts

```diff
@@ -137,3 +137,9 @@ export const DEFI_PROVIDER_TO_METADATA: Record<DefiProvider, DefiProviderMetadat
     url: 'https://app.shapeshift.com',
   },
 }
+
+export const isRunePoolOpportunity = (
+  opportunity: any,
+): opportunity is RunepoolUserStakingOpportunity => {
+  return opportunity.provider === DefiProvider.ThorchainSavers && opportunity.maturity
```

Comment:
q: Will this evaluate to `false` without any user staking? If so, we may be doing some maturity checks on an inactive address resulting in overfetching and possibly unhandled 404s
suggestion: use `thorchainAssetId` as a discriminator

Reply:
Fixed in [4abc315](https://github.com/shapeshift/web/pull/7472/commits/4abc3154365f8295c06c2fb383b10d762be64282)

Reply:
Lovely, ty ser!

Reply:
stylistic if you feel like it: This isn't a constant - can we colocate it next to `UserStakingOpportunity` i.e in its `./utils/index.ts`?

Reply:
Done in [231aabd](https://github.com/shapeshift/web/pull/7472/commits/231aabd2822d7bf58613c7fd32ba5873c67846a8)